### PR TITLE
Fix 50/50 unit test

### PR DESCRIPTION
**Pull request by sahma19**
Flaky Unit test, being fixed, so it isn't blocking Ratatosk

### CODE REVIEW

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs [hunk 1]
```diff
@@ -80,25 +88,25 @@ public async Task WhenCalled_CacheIsSetToHealthy()
...
-                ReceiverTin: "12345678",
-                SenderId: Guid.NewGuid().ToString(),
-                StartDate: DateTimeOffset.Now.ToUnixTimeSeconds()
-            )
-        };
+        memoryCache.Cache.Get(HealthEntries.Key).Should().BeNull("Cache should initially be empty");
 
-        mockHttpMessageHandler.Expect("/api/all-transfer-agreements").Respond("application/json",
+        var agreements = new List<TransferAgreementDto>
+    {
```
Comment (ckr123) [inline domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs:94]: some indention gone wrong here

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs [hunk 2]
```diff
@@ -80,25 +88,25 @@ public async Task WhenCalled_CacheIsSetToHealthy()
...
+
+        mockHttpMessageHandler.Expect("/api/internal-transfer-agreements/all").Respond("application/json",
             JsonSerializer.Serialize(new TransferAgreementsDto(agreements)));
 
-        using var cts = new CancellationTokenSource();
-
         poWalletServiceMock
             .When(x => x.TransferCertificates(Arg.Any<TransferAgreementDto>()))
-            .Do(_ => { cts.Cancel(); });
+            .Do(_ => { /* Simulate transfer without cancelling the token */ });
```
Comment (ckr123) [inline domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs:109]: Why remove this? Wont et resolve in going on forever?

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs [hunk 3]
```diff
@@ -108,13 +116,29 @@ public async Task WhenCalled_CacheIsSetToHealthy()
...
+            metricsMock,
+            memoryCache,
+            serviceProviderMock,
+            httpFactoryMock,
+            transferOptions
+            );
 
-        var act = async () => await worker.StartAsync(cts.Token);
-        await act.Invoke();
+        var cts = new CancellationTokenSource();
```
Comment (ckr123) [inline domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs:128]: Why all this?

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs (hunk 2 above)
Comment (sahma19) [reply domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs:109]: The call to TransferCertificates resulted in cancellation of the entire operation.

This approach is generally more aligned with typical usage patterns, where a service method like TransferCertificates would do its job and return normally, without externally influencing the cancellation state of the entire process.

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs (hunk 3 above)
Comment (sahma19) [reply domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs:128]: 1. Start the worker with worker.StartAsync(cts.Token)

2. await Task.Delay(1000) to allow the worker to perform some initial operations

3. Then, cancel the worker (cts.Cancel()) and wait for it to acknowledge the cancellation and finish its execution (await workerTask)

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs (hunk 1 above)
Comment (sahma19) [reply domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs]: Fixed:

https://github.com/Energinet-DataHub/energy-origin/pull/562/commits/063abd14d904908a90f1ed90dc82d0b875b056c6

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs (hunk 2 above)
Comment (ckr123) [reply domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs:109]: It is intented to cancle the operation as soon as possible. We only test the memorycache here.

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs (hunk 3 above)
Comment (ckr123) [reply domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs:128]: It did that already with         
var act = async () => await worker.StartAsync(cts.Token);
await act.Invoke();

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs [hunk 4]
```diff
@@ -27,18 +27,20 @@ public async Task WhenFail_CacheIsSetToUnhealthy()
         var mockHttpMessageHandler = new MockHttpMessageHandler();
         var memoryCache = new AutomationCache();
 
+        memoryCache.Cache.Get(HealthEntries.Key).Should().BeNull("Cache should initially be empty");
```
Comment (ckr123) [inline domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs:30]: Is there a need to do an assert on the memocache here? 
I would argue that it testing logic in the .NET memory cache and it is out of context of this test

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs (hunk 3 above)
Comment (sahma19) [reply domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs]: Changed back to original implementation, but with correct Uri, and without using in cancellationtoken: b923fa3

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs (hunk 2 above)
Comment (sahma19) [reply domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs]: Changed back to original implementation, but with correct Uri, and without using in cancellationtoken: b923fa3

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs (hunk 4 above)
Comment (sahma19) [reply domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs:30]: Removed.

File: domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs [hunk 5]
```diff
@@ -91,14 +99,17 @@ public async Task WhenCalled_CacheIsSetToHealthy()
...
 
-        using var cts = new CancellationTokenSource();
+        var cts = new CancellationTokenSource();
 
         poWalletServiceMock
             .When(x => x.TransferCertificates(Arg.Any<TransferAgreementDto>()))
-            .Do(_ => { cts.Cancel(); });
+            .Do(_ =>
+            {
+                cts.Cancel(); cts.Dispose();
```
Comment (ckr123) [inline domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs:111]: Should be two lines

**Review by ckr123 — APPROVED** [review]

**Pull request merged**